server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -135,27 +131,52 @@ private void countDown() {
             remoteRequest.runtimeFields(request.runtimeFields());
             remoteRequest.indexFilter(request.indexFilter());
             remoteRequest.nowInMillis(nowInMillis);
-            remoteClusterClient.fieldCaps(remoteRequest,
-                ActionListener.wrap(response -> {
-                    List<FieldCapabilitiesIndexResponse> remotes = new ArrayList<>();
-                    for (FieldCapabilitiesIndexResponse resp : response.getIndexResponses()) {
-                        remotes.add(new FieldCapabilitiesIndexResponse(
-                            RemoteClusterAware.buildRemoteIndexName(clusterAlias, resp.getIndexName()),
-                            resp.get(), resp.canMatch()));
+            remoteClusterClient.fieldCaps(remoteRequest, ActionListener.wrap(response -> {
+                List<FieldCapabilitiesIndexResponse> remotes = new ArrayList<>();
+                for (FieldCapabilitiesIndexResponse resp : response.getIndexResponses()) {
+                    if (resp.getException() == null) {
```

Comment:
Same here, we can simply add the failures map of the remote response to the local one.